packages/utilities/src/controlled.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Determines whether a component is controlled.
+ * @param props - Component props
+ * @param valueProp - Prop containing the controlled value
+ * @param defaultValueProp - Prop containing the uncontrolled default value
+ * @returns true if controlled, false if uncontrolled, undefined if indeterminate (neither value
+ * nor default value prop is set; typically treated as uncontrolled)
```

Comment:
In what cases would callers of this method behave differently in `false` vs `undefined` cases? Can it be explicitly false? It feels weird to use a `boolean` when there are explicitly 3 return values; can this be an enum or flag return instead?

Reply:
I had a reason for doing it this way, but the comment made me think things through again and decide to switch to using just true/false. This more closely matches React's behavior and is more likely to catch subtle issues.